src/extensions/events.py

```diff
@@ -0,0 +1,24 @@
+import discord
+from discord.ext import commands
+import re
+
+class Events:
+  def __init__(self, bot):
+    self.bot: commands.Bot = Bot
+  
+  @commands.event
+  async def on_message(self, message):
+    """
+    Checks for Discord server links in a message
+    then kicks members if links are found
+    """
+    if message.author != self.bot.user:
+      if 'Moderator' not in message.author.roles:
+        pattern = 'discord.gg/[a-z0-9A-Z]+'
+         if re.search(pattern, message.content)
+            await self.bot.delete_message(message)
+            await self.bot.send_message(message.author, 'Advertising is not allowed without permission!')
+            await self.bot.kick(message.author)
```

Comment:
Put this code in a shared method that is called here and in the `?kick` command, or invoke the `kick` method directly.

Use the private message text as the reason for the kick.